### PR TITLE
add camera bounds to the camera component

### LINKED ISSUE #43: Add camera upper and lower bounds in both directions

### PR DESCRIPTION
**Pull request by tlein**
Resolves #43

### CODE REVIEW

File: Engine/Src/Ancona/Core2D/Systems/CameraSystem.cpp [hunk 1]
```diff
@@ -42,7 +44,29 @@ void CameraComponent::MoveCamera()
                 _followPosition->position().x,
                 _followPosition->position().y);
     }
-    _view.setCenter(_view.getCenter() + _offset);
+    BoundCamera();
+    _view.setCenter(round(_view.getCenter().x), round(_view.getCenter().y));
```
Comment (jeffswenson) [inline Engine/Src/Ancona/Core2D/Systems/CameraSystem.cpp]: You removed the ability to offset.

File: Engine/Src/Ancona/Core2D/Systems/CameraSystem.cpp [hunk 2]
```diff
@@ -42,7 +44,29 @@ void CameraComponent::MoveCamera()
...
+}
+
+void CameraComponent::BoundCamera()
+{
+    if (_view.getCenter().x > _xBounds.y)
```
Comment (jeffswenson) [inline Engine/Src/Ancona/Core2D/Systems/CameraSystem.cpp]: This code should be more functional. It should include the rounding and offset. You could then get rid of some of the conditions using std::max(std::min(currentY, maxY), minY);

Comment (tlein) [reply Engine/Src/Ancona/Core2D/Systems/CameraSystem.cpp]: fixed

File: Engine/Src/Ancona/Core2D/Systems/CameraSystem.cpp (hunk 1 above)
Comment (tlein) [reply Engine/Src/Ancona/Core2D/Systems/CameraSystem.cpp]: fixed

**Pull request merged**